### PR TITLE
feat(ci): Adds azure devops pipeline files

### PR DESCRIPTION
**Pull request by thomastaylor312**

### CODE REVIEW

**Comment by thomastaylor312** [comment]
Please note that the existing GH actions pipelines will not run since I deleted them. I will remove those check requirements once we merge

File: .ci/provider.yml [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+  - name: WASH_ISSUER_KEY
+    value: $[variables.COSMONIC_ACCOUNT_OFFICIAL]
+  - name: WASH_SUBJECT_KEY
+    value: $[variables.COSMONIC_KUBERNETES_APPLIER_KEY]
+
+stages:
+  - stage: build_and_check
+    jobs:
+      - job: build_and_check
+        # TODO: Build on all OS types. This will require a working wash install on all OS
```
Comment (thomastaylor312) [inline .ci/provider.yml:45]: I will handle this in a follow up PR

File: .ci/actor.yml [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+
+  - stage: release
+    dependsOn:
+      - build_and_check
+    # Only do this stage if the ref is a tag and the previous stage succeeded
+    condition: and(succeeded('build_and_check'), startsWith(variables['Build.SourceBranch'], 'refs/tags/'))
+    jobs:
+      - job: release
+        steps:
+          - template: steps/install-wash-ubuntu.yml@public-templates
```
Comment (connorsmith256) [inline .ci/actor.yml:79]: This got me thinking, if the release stage only runs after the build stage, do we need to install wash/rust toolchain stuff again?

**Review by connorsmith256 — APPROVED** [review]
LGTM

File: .ci/actor.yml (hunk 2 above)
Comment (thomastaylor312) [reply .ci/actor.yml:79]: Yeah because I believe each stage runs on a clean machine. Have not verified that though

Comment (connorsmith256) [reply .ci/actor.yml:79]: You're right 👍 Steps share the same filesystem/agent, but apparently jobs and stages don't share anything

**Pull request merged**